src/macaron/slsa_analyzer/package_registry/npm_registry.py

```diff
@@ -199,6 +201,47 @@ def download_attestation_payload(self, url: str, download_path: str) -> bool:
 
         return False
 
+    def get_latest_version(self, namespace: str, name: str) -> str | None:
+        """Try to retrieve the latest version of a package from the registry.
+
+        Parameters
+        ----------
+        namespace: str
+            The optional namespace of the package.
+        name: str
+            The name of the package.
+
+        Returns
+        -------
+        str | None
+            The latest version of the package, or None if one cannot be found.
+        """
+        if not name:
+            return None
+
+        url = f"https://{self.hostname}"
+        if namespace:
+            url = f"{url}/{namespace}"
+        url = f"{url}/{name}/latest"
+
+        response = send_get_http_raw(url, timeout=self.request_timeout)
+
+        if not response or not response.text:
+            logger.debug("No valid response from NPM server for latest version.")
+            return None
+
+        json_data = json.loads(response.text)
+        try:
+            version = json_data["version"]
+        except KeyError:
+            version = ""
```

Comment:
Using `.get` is much nicer than trying to access with the `[]` operator and catching `KeyError`.
```suggestion
        version = json_data.get("version")
```